source/simulator/core.cc

```diff
@@ -152,6 +153,7 @@ namespace aspect
     parameters (prm, mpi_communicator_),
     melt_handler (parameters.include_melt_transport ? new MeltHandler<dim> (prm) : NULL),
     newton_handler (parameters.nonlinear_solver == NonlinearSolver::Newton_Stokes ? new NewtonHandler<dim> () : NULL),
+    world_builder (parameters.world_builder_directory != "" ? WorldBuilder::create_world_builder<dim> (prm) : NULL),
```

Comment:
I would always create such an object. It may simply not be used if no plugin references it.

The presence of the `world_builder_directory` variable in the main class confuses me. What is it used for? Why is that not a parameter of the world builder itself, and located in the corresponding structure?

Reply:
>I would always create such an object. It may simply not be used if no plugin references it.

sure.

>The presence of the `world_builder_directory` variable in the main class confuses me. What is it used for? Why is that not a parameter of the world builder itself, and located in the corresponding structure?

It should have been `world_builder_file`. But now I won't need it now we always create the world builder, so I removed it